ldpd/lde.c

```diff
@@ -102,9 +118,28 @@ static struct quagga_signal_t lde_signals[] =
 	},
 };
 
+struct zclient *zclient_sync = NULL;
+static void
+zclient_sync_init (u_short instance)
+{
+	/* Initialize special zclient for synchronous message exchanges. */
+	log_debug ("Initializing synchronous zclient for label manager");
+	zclient_sync = zclient_new (master);
+	zclient_sync->sock = -1;
+	zclient_sync->redist_default = ZEBRA_ROUTE_LDP;
+	zclient_sync->instance = instance;
+	if (zclient_socket_connect (zclient_sync) < 0) {
+		log_warn ("Error connecting synchronous zclient!");
+		exit (1);
```

Comment:
ldpd should not die here as we should allow it to start before zebra and work in the same way.

pimd also uses a synchronous zclient and tries to connect to zebra over and over again until it succeeds. Please take a look at the zclient_lookup_connect() function in pimd/pim_zlookup.c.